feat: upstream merge 04 10 23

abocsan-plenty opened the pull request:
## Why:

Closes: #ID

## Describe your changes

## Checklist before requesting a review
- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

ksted commented on packages/composables/src/getters/categoryGetters.ts:
```diff
@@ -1,99 +1,20 @@
-    count: category?.itemCount[0]?.count || 0
-  };
+interface categoryGetters {
+  getCategoryName(category: Category),
+  getCategoryDetails(category: Category)
```
```suggestion
  getCategoryName(category: Category): string,
  getCategoryDetails(category: Category): CategoryDetails | null
```
**todo**: Add return types.

maxiroellplenty replied:
done

ksted commented on packages/api-client/src/types/category.ts:
```diff
@@ -1,20 +1,52 @@
-    image2Path: string,
-};
+export interface CategoryDetails {
+    metaRobots: string
+    shortDescription: string
+    singleItemView: string
+    canonicalLink: string
+    fulltext: string
+    updatedBy: string
+    metaDescription: string
+    metaTitle: string
+    lang: string
+    name: string
+    categoryId: string
+    itemListView: string
+    position: string
+    description2: string
+    nameUrl: string
+    plenty_category_details_image_path: string
+    plenty_category_details_image2_path: string
+    description: string
+    updatedAt: string
+    metaKeywords: string
+    pageView: string
+    image: number
+    imagePath: string
+    image2: string
+    image2Path: string
+    plentyId: number
```
```suggestion
    canonicalLink: string
    categoryId: string
    description: string
    description2: string
    fulltext: string
    image: number
    imagePath: string
    image2: string
    image2Path: string
    itemListView: string
    lang: string
    metaDescription: string
    metaKeywords: string
    metaRobots: string
    metaTitle: string
    name: string
    nameUrl: string
    pageView: string
    plenty_category_details_image_path: string
    plenty_category_details_image2_path: string
    plentyId: number
    position: string
    shortDescription: string
    singleItemView: string
    updatedAt: string
    updatedBy: string
```
**suggestion**: Improve readability of long list by sorting the entries alphabetically. `image`, `image2`, `image2Path` and `imagePath` is an exception because of the logical connection.

As a side note, there's a [VSC extension](https://marketplace.visualstudio.com/items?itemName=zhihaocui.ts-interface-sorter) for automatically sorting interfaces.

maxiroellplenty replied:
done

maxiroellplenty commented on packages/composables/src/getters/categoryTreeGetters.ts:
```diff
@@ -0,0 +1,93 @@
+import { CategoryGetters, AgnosticCategoryTree, AgnosticBreadcrumb } from '@vue-storefront/core';
+import type { CategoryTreeItemDetails, CategoryTreeItem } from '@vue-storefront/plentymarkets-api';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
done

maxiroellplenty commented on packages/composables/src/getters/categoryTreeGetters.ts:
```diff
@@ -0,0 +1,93 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function getTree(category: CategoryTreeItem): AgnosticCategoryTree {
+  return {
+    label: getCategoryDetails(category.details)?.name || '',
+    slug: getCategoryDetails(category.details)?.nameUrl || '',
```
done

MaciejWolowski committed:
feat: add some tests and fix others (#118)

feat: adding tests for new pages

added tests for new pages
changed some configuration for cypress
add visitandhydrate method

MaciejWolowski committed:
feat: fix productlist tests [SV-207] (#132)

* feat: adding tests for new pages

added tests for new pages
changed some configuration for cypress
add visitandhydrate method

* feat: update product list tests

renovate[bot] committed:
chore(deps): :arrow_up: Update definitelyTyped (#134)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): :arrow_up: Update devDependency eslint to v8.49.0 (#138)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): :arrow_up: Update devDependency @types/node to v18.17.15 (#137)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

MaciejWolowski committed:
feat: add tests for my account page [SV-208] (#131)

* feat: add tests for my account page

* feat: testing solution

* Update apps/web/__tests__/support/pageObjects/MyAccountPageObject.ts

Co-authored-by: Szymon Dziewoński <szymon.dziewonski@gmail.com>

* feat: added 2 lines for my account test

* feat: test fix for test

* feat: test fix

---------

Co-authored-by: Szymon Dziewoński <szymon.dziewonski@gmail.com>

dawid-ziobro committed:
fix: remove logout button on my account page for larger devices (#139)

dawid-ziobro committed:
chore: update sonarcloud ci (#140)

dawid-ziobro committed:
feat: add useBreadcrumbs composable (#142)

dawid-ziobro committed:
refactor: align code (#143)

justyna-13 committed:
chore: unit tests workflow [SV-214] (#141)

Co-authored-by: Szymon Dziewoński <szymon.dziewonski@gmail.com>

dawid-ziobro committed:
fix: replace password inputs with dedicated component (#147)

renovate[bot] committed:
chore(deps): :arrow_up: Update devDependency @types/node to v18.17.17 (#144)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): :arrow_up: Update definitelyTyped (#151)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): :arrow_up: Update devDependency eslint to v8.50.0 (#152)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

FRSgit committed:
fix: remove createSharedComposable from sdk (#153)

Signed-off-by: Jakub Freisler <jakub@frsource.org>

renovate[bot] committed:
chore(deps): :arrow_up: Update devDependency prettier to v3 (#146)

* chore(deps): :arrow_up: Update devDependency prettier to v3

* chore: changes connected to major bump update

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Szymon Dziewoński <szymon.dziewonski@gmail.com>

filrak committed:
docs: update README.md with new docs links (#150)

Update README.md

renovate[bot] committed:
chore(deps): :arrow_up: Update all minor and patch dependency bump (#133)

* chore(deps): :arrow_up: Update all minor and patch dependency bump

* chore: manual changes

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Szymon Dziewoński <szymon.dziewonski@gmail.com>

renovate[bot] committed:
chore(deps): :arrow_up: Update all major dependency bump (major) (#145)

* chore(deps): :arrow_up: Update all major dependency bump

* chore: manual adjustments

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Szymon Dziewoński <szymon.dziewonski@gmail.com>

renovate[bot] committed:
chore(deps): :arrow_up: Update devDependency prettier to v3.0.3 (#155)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

abocsan-plenty committed:
Merge remote-tracking branch 'upstream/develop' into feat/upstream-merge-04-10-23

abocsan-plenty committed:
feat: upstream merge 04-10-23

abocsan-plenty committed:
feat: upstream merge 04-10-23

abocsan-plenty committed:
chore: commennt added test

abocsan-plenty committed:
Merge branch 'main' into feat/upstream-merge-04-10-23

abocsan-plenty committed:
chore: lint fixes

abocsan-plenty committed:
Merge remote-tracking branch 'origin/feat/upstream-merge-04-10-23' into feat/upstream-merge-04-10-23

abocsan-plenty committed:
chore: unit tests fix

abocsan-plenty committed:
chore: lint fixes

abocsan-plenty committed:
chore: suppress unit test

maxiroellplenty commented on apps/web/components/CategoryPageContent/CategoryPageContent.vue:
```diff
@@ -83,19 +83,11 @@ const runtimeConfig = useRuntimeConfig();
-    close();
-  }
-});
+whenever(isTablet, close);
```
thats cool :D

maxiroellplenty replied:
dont know why its better than just the watch but hey

maxiroellplenty committed:
Merge branch 'main' into feat/upstream-merge-04-10-23
